src/detect-filestore.c

```diff
@@ -207,11 +233,36 @@ static int DetectFilestorePostMatch(DetectEngineThreadCtx *det_ctx,
 {
     SCEnter();
 
+    if (p->flow == NULL) {
+#ifndef DEBUG
+        SCReturnInt(0);
+#else
+        BUG_ON(1);
+#endif
+    }
+
     if (det_ctx->filestore_cnt == 0) {
+        if (s->filestore_ctx && (s->filestore_ctx->scope == FILESTORE_SCOPE_SSN)) {
```

Comment:
I need some comment to understand

For me `if (det_ctx->filestore_cnt == 0)` means we do not have to store anything as all the rules with `filestore` were tried and none matched

Reply:
Adding a comment, we need to do something if we have a scope of TCP session (via filestore option) as we need to capture file that can appear later.